rpc_jobs/rpc_artifact_build.yml

```diff
@@ -131,43 +79,136 @@
+          python_parallel = [:]
+          for (x in image_list[env.series]) {{
+            // Need to bind the image variable before the closure - can't do 'for (image in ...)'
+            // https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
+            def image = x
+            python_parallel[image] = {{
+              artifact_build.python(image)
+            }}
+          }}
+
+          // Container artifacts can be built in parallel for all
+          // distributions (images) supported by a series. This
+          // can currently only be done in parallel for PR's due
+          // to a race condition when updating the image index.
+          // ref: RE-115
+          container_parallel = [:]
+          for (x in image_list[env.series]) {{
+            // Need to bind the image variable before the closure - can't do 'for (image in ...)'
+            // https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
+            def image = x
+            container_parallel[image] = {{
+              artifact_build.container(image)
+            }}
+          }}
+
+          // We can run all the artifact build processes in parallel
+          // for PR's because they do not upload anything.
+          if (env.trigger == "pr") {{
+            parallel(
+              "apt": {{
+                // We do not need to download existing data from
+                // rpc-repo for PR's as they do not upload their
+                // changes. We only need to do that when making
+                // changes to the data. Skipping the data sync
+                // for PR's saves a ton of time.
+                env.PULL_FROM_MIRROR = "NO"
                 artifact_build.apt()
-              }} // node ArtifactBuilder2
-            }} // stage
-          ) // conditionalStage
-          // continuing on CIT slave
-          artifact_build.git()
-          artifact_build.python()
-          artifact_build.container()
+              }},
+              "git": {{
+                // We only care about building these on one of the
+                // images because the artifacts produced are not
+                // distro-specific. We take the first one for
+                // convenience.
+                artifact_build.git(image_list[env.series][0])
+              }},
+              "python": {{
+                parallel python_parallel
+              }},
+              "container": {{
+                parallel container_parallel
+              }}
+            ) // parallel
+          }} else {{
+            // When the job is periodic, the jobs must run in a
+            // particular sequence in order to ensure that each
+            // artifact set is built and uploaded so that it can
+            // be used in the step that follows.
+
+            // The apt artifacts are built on a long-lived slave
+            // with a single executor, so no locking is required.
+            artifact_build.apt()
```

Comment:
I think @mattt416 has a point about stages. For PRs, each build is in a labeled block as each parallel section has a label. However in the periodic/serialised build, we've removed the conditional stages and not added any fixed stages. Adding these would be helpful for finding the correct sections of logs, and correctly labeling the blue ocean flow chart. 

All four blocks in the periodic section should be wrapped in a stage. Stage should be the outermost call.